rootfs/scheduler/resources/pod.py

```diff
@@ -719,7 +719,41 @@ def _handle_not_ready_pods(self, namespace, labels):
                     message = "\n".join([x.strip() for x in event['message'].split("\n")])
                     raise KubeException(message)
 
-        return None
+        # Handle cases when no one pods is started (quota check fialed)
+        # If quota is enabled in a namespace for compute resources like cpu and
+        # memory, users must specify requests or limits for those values;
+        # otherwise, the quota system may reject pod creation.
+        # http://kubernetes.io/docs/admin/resourcequota/
+        # or when scale process trigger overuse case for some pods:
+        # For example: used 3/3 pods, but user scale to 4
+        events = self._get_namespace_events(namespace)
+        if len(events) != 0:
+            log = self._get_formatted_messages(events)
+            self.log(namespace, log)
+            raise KubeException(log)
```

Comment:
> And i have no idea, how to get last RS for specified deployment in deis controller.

[The unit tests](https://github.com/deis/controller/blob/890a26318802fedf1ce82c42ad2401af57671af4/rootfs/scheduler/tests/test_deployments.py#L188) demonstrate how one can get a replicaset deployed by the controller.

> And I actually don't understand why we need to return None when required pods != actual pods during operation.

I'm sorry, I don't understand what you mean by this. Is there something in the code that needs clarifying, and if so where? The only added condition in this PR where I see we return `None` is when no events have occurred in the namespace on line 735.

> I can add regexp to detect patterns with "exceeded quota", but this message may changes in next releases.

If you're checking the replicaset itself, you can probably just check it for a condition called ["ReplicaFailure"](https://github.com/kubernetes/client-go/blob/8fa0506b2636f5a0324d0cd7783efacc5edb2464/pkg/apis/extensions/v1beta1/types.go#L890-L893) instead, which is a catch-all for quotas, limits, security policies, node selectors etc. That should probably live in [`deployments.are_replicas_ready()`](https://github.com/deis/controller/blob/890a26318802fedf1ce82c42ad2401af57671af4/rootfs/scheduler/resources/deployment.py#L242), though.